### PR TITLE
186786001 spdat

### PR DESCRIPTION
**Pull request by ttoomey**
## Description

* support helper text/html on choice options
* support numeric value / score on choice options
* support simple interpolation in display component

[support assessment score
](https://www.pivotaltracker.com/story/show/186786000)
[support spdat
](https://www.pivotaltracker.com/story/show/186786001)

[//]: # 'remove if not applicable'
[Depends on hmis-warehouse PR:
](https://github.com/greenriver/hmis-warehouse/pull/3954)
## Type of change
[//]: # 'remove options that are not relevant'
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/components/elements/input/RadioGroupInput.tsx [hunk 1]
```diff
@@ -145,34 +146,40 @@ const RadioGroupInput = ({
...
+          {options.map(({ code, label, helperText }) => (
+            <Fragment key={code}>
+              <FormControlLabel
+                data-testid={`option-${code}`}
+                disabled={props.disabled}
+                value={code}
+                aria-label={label || code}
+                onClick={(e) => onClickOption(e, code)}
+                control={
+                  <ControlComponent
```
Comment (gigxz) [inline src/components/elements/input/RadioGroupInput.tsx]: For accessibility this relies on being rendered inside the label. But we should probably link up the helper text to it too. I think something like`aria-labelledby="main-label-element helper-text-element>"` here?

File: src/modules/form/util/formUtil.ts [hunk 2]
```diff
@@ -453,6 +454,22 @@ export const shouldEnableItem = ({
   }
 };
 
+const isPicklistOption = (
```
Comment (gigxz) [inline src/modules/form/util/formUtil.ts]: We already have a type guard function called isPickListOption -- did you miss that, or did you need a separate one for the fragment?

**Review by gigxz — COMMENTED** [review]
Looks good! a couple small comments

File: src/components/elements/input/RadioGroupInput.tsx (hunk 1 above)
Comment (ttoomey) [reply src/components/elements/input/RadioGroupInput.tsx]: Good suggestion, I did a little work to improve accessibility of the input component; I also extracted the option out into its own component.
* removed rid of the enclosing MUI Form Group (it didn't seem to be needed)
* add fieldset and legend elements for screen readers
* add aria-labelledby for both label and helperText. (I suspect aria-describedby would be more accurate but I couldn't get that work with voiceover)

File: src/modules/form/util/formUtil.ts (hunk 2 above)
Comment (ttoomey) [reply src/modules/form/util/formUtil.ts]: ah I missed that, thanks

**Review by gigxz — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**